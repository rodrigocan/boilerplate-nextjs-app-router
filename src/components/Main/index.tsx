const Main = () => (
  <main className="bg-main text-white w-full h-full p-12 text-center flex flex-col items-center justify-center">
    <h1>React Avan√ßado</h1>
  </main>
)

export default Main
