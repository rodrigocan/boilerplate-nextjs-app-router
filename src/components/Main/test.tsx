import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderiza o componente
    const { container } = render(<Main />)

    // busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    // renderiza o componente
    const { container } = render(<Main />)

    // verifica se o background-color está correto
    // expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
    console.log(container.firstChild)
  })
})
