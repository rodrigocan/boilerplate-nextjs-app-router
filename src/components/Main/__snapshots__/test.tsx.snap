// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Main /> > should render the heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
